plugins/flytekit-airflow/flytekitplugins/airflow/task.py

```diff
@@ -1,51 +1,175 @@
+    These tasks don't have async method to get the job status, so cannot be used in the Flyte agent. We run these tasks in a container.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        task_config: AirflowObj,
+        inputs: Optional[Dict[str, Type]] = None,
+        **kwargs,
+    ):
+        super().__init__(
+            name=name,
+            task_config=task_config,
+            interface=Interface(inputs=inputs or {}),
+            **kwargs,
+        )
+        self._task_resolver = airflow_task_resolver
+
+    def execute(self, **kwargs) -> Any:
+        logger.info("Executing Airflow task")
+        _get_airflow_instance(self.task_config).execute(context=Context())
+
 
+class AirflowTask(AsyncAgentExecutorMixin, PythonTask[AirflowObj]):
+    """
+    This python task is used to wrap an Airflow task. It is used to run an Airflow task in Flyte agent.
+    The airflow task module, name and parameters are stored in the task config. We run the Airflow task in the agent.
+    """
 
-class AirflowTask(AsyncAgentExecutorMixin, PythonTask[AirflowConfig]):
     _TASK_TYPE = "airflow"
 
     def __init__(
         self,
         name: str,
-        query_template: str,
-        task_config: Optional[AirflowConfig],
+        task_config: Optional[AirflowObj],
         inputs: Optional[Dict[str, Type]] = None,
         **kwargs,
     ):
         super().__init__(
             name=name,
             task_config=task_config,
-            query_template=query_template,
             interface=Interface(inputs=inputs or {}),
             task_type=self._TASK_TYPE,
             **kwargs,
         )
 
     def get_custom(self, settings: SerializationSettings) -> Dict[str, Any]:
+        # Use jsonpickle to serialize the Airflow task config since the return value should be json serializable.
         return {"task_config_pkl": jsonpickle.encode(self.task_config)}
 
 
+def _get_airflow_instance(airflow_obj: AirflowObj) -> typing.Union[BaseOperator, BaseSensorOperator, BaseTrigger]:
+    # Set the GET_ORIGINAL_TASK attribute to True so that obj_def will return the original
+    # airflow task instead of the Flyte task.
+    ctx = FlyteContextManager.current_context()
+    ctx.user_space_params.builder().add_attr("GET_ORIGINAL_TASK", True).build()
+
+    obj_module = importlib.import_module(name=airflow_obj.module)
+    obj_def = getattr(obj_module, airflow_obj.name)
+    if (
+        issubclass(obj_def, BaseOperator)
+        and not issubclass(obj_def, BaseSensorOperator)
+        and not _is_dataflow_operator(obj_def)
+    ):
+        try:
+            return obj_def(**airflow_obj.parameters, deferrable=True)
+        except AirflowException:
+            logger.debug(f"Airflow operator {airflow_obj.name} does not support deferring")
+
+    return obj_def(**airflow_obj.parameters)
+
+
+def _is_dataflow_operator(cls: Type):
```

Comment:
```suggestion
def _is_not_supported(cls: Type):
```
Or can be called `_is_not_agent_supported`